fiona/_err.pyx

```diff
@@ -227,7 +234,9 @@ cdef void *exc_wrap_pointer(void *ptr) except NULL:
         exc = exc_check()
         if exc:
             raise exc
-            return NULL
+        else:
+            # null pointer was passed, but no error message from GDAL
+            raise FionaNullPointerError(-1, -1, "NULL pointer error")
```

Comment:
Either the material at http://cython.readthedocs.io/en/latest/src/reference/language_basics.html#error-and-exception-handling has changed or I completely misread this part

> Functions declared with an exception value, should explicitly prevent a return of that value.